builder/src/prepare_project.py

```diff
@@ -151,7 +291,263 @@ def supports(self, version):
+                  'this time.').format(minor_version)
+            sys.exit(1)
+
+        project_name = self._get_project_assembly_name()
+        assembly_name = ASSEMBLY_NAME_TEMPLATE.format(project_name)
+        if not os.path.isfile(os.path.join(self.root, assembly_name)):
+            print('Cannot find entry point assembly {0} for ASP.NET Core project'
+                  .format(assembly_name))
+            sys.exit(1)
+
+        contents = PublishedApp.DOCKERFILE_CONTENTS.format(runtime_image=base_image.image,
+                                                           dll_name=project_name)
+        with open(output, 'wt') as out:
+            out.write(contents)
+
+    def _get_project_assembly_name(self):
+        """Returns the name of the entrypoint assembly given the .deps.json
+        file name.
+
+        Returns:
+            The name of the entry point assembly.
+        """
+        filename = os.path.basename(self.deps_path)
+        return filename[:-len(DEPS_EXTENSION)]
+
+    def _get_runtime_minor_version(self):
+        """Determines the target of the .NET Core runtime needed by the app.
+
+        Reads the given .deps.json file and determines the version of the
+        runtime used by the app.
+
+        Returns:
+            The version of the runtime used by the app.
+        """
+        with open(self.deps_path, 'r') as src:
+            content = json.load(src)
+            try:
+                libraries = content['libraries']
+                for key in libraries:
+                    if key.lower().startswith(NETCORE_APP_PREFIX):
+                        version = key[len(NETCORE_APP_PREFIX):]
+                        return version.split('-')[0]
+            except KeyError:
+                return None
+
+
+class SingleProjectApp(object):
+    """An app that is composed of a single .csproj build file.
+
+    This class represents an app that is composed of a single app, as
+    it is with the apps generated using "dotnet new" by default.
+
+    Since there is a single project, this project is considered to be
+    the entry point for the app.
+    """
+
+    # Dockerfile template to be used when packaging .csproj based apps.
+    DOCKERFILE_CONTENTS = textwrap.dedent(
+        """\
+        FROM gcr.io/cloud-builders/csharp/dotnet AS builder
+        COPY . /src
+        WORKDIR /src
+        RUN dotnet restore --packages /packages
+        RUN dotnet publish -c Release -o /published
+
+        FROM {runtime_image}
+        COPY --from=builder /published /app
+        ENV ASPNETCORE_URLS=http://*:${{PORT}}
+        WORKDIR /app
+        ENTRYPOINT [ "dotnet", "{dll_name}.dll" ]
+        """)
+
+    def __init__(self, root, project):
+        """Initializes the instance of SingleProjectApp.
+
+        Args:
+            root: A string with the path to the directory that contains the project.
+            project: A string with the path to the project file.
+        """
+        self.root = root
```

Comment:
`self.root` does not seem to be used anywhere.

Reply:
Donte.